File: euclidean_gcd/EuclideanGCD.cs
using System;

public class EuclideanGCD
{
    public static long euclidean_gcd(long a, long b)
    {
        if(a == 0)
        {
            return (b);
        }
        else
        {
            return euclidean_gcd(b % a, a);
        }
    }

    public static long euclidean_gcd_iterative(long a, long b)
    {
        while (b != 0)
        {
            long temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }

    public static void Main()
    {
        int a = 9000, b = 145685;
        Console.WriteLine("GCD of " + a + " and " + b + " by recursive is : " + euclidean_gcd(a, b));
        Console.WriteLine("GCD of " + a + " and " + b + " by iterative is : " + euclidean_gcd_iterative(a, b));
        Console.WriteLine("");
    }
}

File: euclidean_gcd/EuclideanGCD.cpp
#include <bits/stdc++.h>
using namespace std;

int euclidean_gcd_recursive(int a, int b)
{
    if (a == 0)
        return b;
    return euclidean_gcd_recursive(b % a, a);
}

int euclidean_gcd_iterative(int a, int b) {
	while(b != 0) {  
		int temp = b;  
		b = a % b;
		a = temp;
	}
	return a;
}

int main() {
	int a = 69;
	int b = 11;
	int answer_recursive = euclidean_gcd_recursive(a, b);
	int answer_iterative = euclidean_gcd_iterative(a, b);
	cout<<"GCD of "<< a <<" and "<< b <<" is : "<< answer_recursive<<" by recursive algo"<<endl;
	cout<<"GCD of "<< a <<" and "<< b <<" is : "<<answer_iterative<<" by iterative algo"<<endl;
	return 0;
}
